test/compilable/ddoc_markdown_links.d

```diff
@@ -0,0 +1,33 @@
+// PERMUTE_ARGS:
+// REQUIRED_ARGS: -D -Dd${RESULTS_DIR}/compilable -o- -preview=markdown
+// POST_SCRIPT: compilable/extra-files/ddocAny-postscript.sh
+
+/++
+# Links
+
+[unused reference]: https://nowhere.com
+
+A link to [Object].
+
+A link to [the base object][Object].
+
+Not a link because it's an associative array: int[Object].
+
+An inline link to [the D homepage](https://dlang.org).
+
+A reference link to [the D homepage][d site].
+
+Not a reference link because it links to [nothing][nonexistent].
+
+A simple link to [dub].
```

Comment:
What's the outcome if this both matches a symbol and a reference?

Reply:
References are preferred. I'll add that to the proposed docs.